src/s3/util/messages.ts

```diff
@@ -3,16 +3,69 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { showLogOutputChannel } from '../../shared/logger'
+import * as vscode from 'vscode'
+import { getLogger, showLogOutputChannel } from '../../shared/logger'
 import { localize } from '../../shared/utilities/vsCodeUtils'
 import { Window } from '../../shared/vscode/window'
 
-export function showErrorWithLogs(message: string, window: Window): void {
+/**
+ * Shows a non-modal error message with a button to open the log output channel.
+ *
+ * @returns	A promise that resolves when the button is clicked or the error is dismissed.
+ */
+export async function showErrorWithLogs(message: string, window: Window): Promise<void> {
     const logsItem = localize('AWS.generic.message.viewLogs', 'View Logs...')
 
-    window.showErrorMessage(message, localize('AWS.generic.message.viewLogs', 'View Logs...')).then(selection => {
-        if (selection === logsItem) {
-            showLogOutputChannel()
-        }
-    })
+    return window
+        .showErrorMessage(message, localize('AWS.generic.message.viewLogs', 'View Logs...'))
+        .then(selection => {
+            if (selection === logsItem) {
+                showLogOutputChannel()
+            }
+        })
+}
+
+/**
+ * Shows a modal confirmation (warning) message with buttons to confirm or cancel.
+ *
+ * @param prompt the message to show.
+ * @param confirm the confirmation button text.
+ * @param cancel the cancel button text.
+ * @param window the window.
+ */
+export async function showConfirmationMessage(
```

Comment:
Should this be a modal or a quick pick? For reference, does it make sense to have the confirmation for deleting a file vs. a folder from S3 appear in different locations?

Reply:
You mean file vs. bucket? Ideally I think we'd have a more prominent input box for the super confirmation dialog on bucket deletion. Quick pick doesn't necessarily do justice to the severity of this action.

But since that isn't an option, I think having them different is better than making the file deletion into a quick pick. I think users are used to these sorts of warning popups for severe sync actions.

---

One thing I considered, though, was introducing an additional modal dialog *before* the input box for the bucket deletion. 

The user would first see "Are you sure you want to delete bucket ___?" (Cancel/Delete) 

If the user clicks Delete, only then would they see the input box for entering the bucket name to confirm. I just tried this out quickly and it looks fine to me and could prevent more accidental deletions. What do you think?

Reply:
I'm thinking that it's hard to be accidental about right clicking a field, clicking delete, and then clicking a confirmation (or entering in a text confirmation). It's a severe action, yes, but there's a lot of intent that goes into it.

What this all boils down to, though, is that I'm just not really a fan of a modal since it looks so different from the rest of the UI (vs. a quick pick that won't lose focus). Again, it's not anything blocking, it's just a matter of taste.

Reply:
Understood. I agree it would be nice if vscode/electron could emulate a modal without delegating to the OS to take care of the dialog. 

*Side note: It looks a lot better on mac vs windows since it's attached to the top vs floating resp.*

I do feel quite strongly about the potential for accidental clicks. Delete options are always sitting right below innocuous options like "Copy". Users on auto-pilot (doing something repetitively as part of a manual workflow) could misclick quite easily. I know I've done this before. The confirmation dialog saves you.